internal/controllers/machine/machine_controller_phases.go

```diff
@@ -116,9 +116,8 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, cluster *clusterv1.C
 		return external.ReconcileOutput{}, err
 	}
 
-	// With the migration from v1alpha2 to v1alpha3, Machine controllers should be the owner for the
-	// infra Machines, hence remove any existing machineset controller owner reference
-	if controller := metav1.GetControllerOf(obj); controller != nil && controller.Kind == "MachineSet" {
+	// Ensure the Machine is the controller owner for infrastructure and bootstrap object it references.
```

Comment:
Can we add a comment to explain why this is required/when an infra or bootstrap object might have a controller which isn't the machine itself?